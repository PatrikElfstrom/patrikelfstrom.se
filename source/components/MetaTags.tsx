import { ReactElement } from 'react';
import { Helmet } from 'react-helmet';
import structuredData from '../static/structuredData.json';

export const MetaTags = (): ReactElement => {
  const { origin } = window.location;

  return (
    <Helmet>
      <title>{structuredData.mainEntityOfPage.name}</title>
      <meta name="description" content={structuredData.mainEntityOfPage.description} />
      <link rel="canonical" href={origin} />

      <meta property="og:image" content="/images/icons/patrikelfstrom-logo-1200x1200.png" />
      <meta property="og:image:width" content="1200" />
      <meta property="og:image:height" content="1200" />
      <meta property="og:image:alt" content="Patrik ElfstrÃ¶m logotype" />
      <meta property="og:url" content={origin} />
      <meta property="og:type" content="website" />

      <meta name="twitter:card" content="summary" />

      <link rel="icon" type="image/png" sizes="32x32" href="/images/icons/favicon-32x32.png" />
      <link rel="icon" type="image/png" sizes="16x16" href="/images/icons/favicon-16x16.png" />
      <link rel="mask-icon" href="/images/icons/safari-pinned-tab.svg" color="#0084c7" />

      <link rel="apple-touch-icon" href="/images/icons/apple-touch-icon.png" />
      <meta name="apple-mobile-web-app-capable" content="yes" />
      <meta name="apple-mobile-web-app-status-bar-style" content="default" />
      <meta name="theme-color" content="#ffffff" />

      <link rel="author" href="/humans.txt" />
      <link rel="manifest" href="/manifest.webmanifest" />
      <script type="application/ld+json">{JSON.stringify(structuredData)}</script>
    </Helmet>
  );
};
