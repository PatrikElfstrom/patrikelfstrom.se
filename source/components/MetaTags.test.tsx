import TestRenderer from 'react-test-renderer';
import { MetaTags } from './MetaTags';

describe('<MetaTags />', () => {
  it('should match snapshot', () => {
    const tree = TestRenderer.create(<MetaTags />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
