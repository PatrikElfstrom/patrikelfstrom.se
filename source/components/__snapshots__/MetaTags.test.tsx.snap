// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MetaTags /> should match snapshot 1`] = `null`;
